src/App.jsx

```diff
@@ -2,14 +2,17 @@ import { AuthModal } from './components/auth'
 import { Header } from './components/header'
 import { Introduction } from './components/introduction'
 import { SongList } from './components/songs/list'
+import { useCheckout } from './hooks/useCheckout'
 
 function App() {
+   const status = useCheckout()
```

Comment:
About, your question about the delay while the application is getting the auth state, yes it is normal, because the application is mounted first so the initial state is unauthenticated, then you are using the useCheckout() custom hook to validate the state and change it from there (this happens second).

But this is not a great user experience, someone with a slow connection may think they are not authenticated, make sure the application loads only when we have the firebase authentication state.